docs/source/pagination/introduction.mdx

```diff
@@ -0,0 +1,123 @@
+    pageResolver: { page, paginationDirection in
+        // As we only want to support forward pagination, we can return `nil` for reverse pagination
+        switch paginationDirection {
+        case .next:
+            return MyQuery(first: 10, after: page.endCursor ?? .none)
+        case .previous:
+            return nil
+        }
+    },
+    extractPageInfo: { data in
+        CursorBasedPagination.Forward(
+            hasNext: data.values.pageInfo.hasNextPage ?? false,
+            endCursor: data.values.pageInfo.endCursor
+        )
+    }
+)
+```
+
+In this example, the `GraphQLQueryPager` instance is initialized with a closure that provides the pagination query to be executed, and another closure that extracts the pagination information from the query result. The `GraphQLQueryPager` instance can then be used to fetch the paginated data, and to watch for changes to the paginated data.
+
+We could similarly support forward offset-based pagination by supplying `OffsetPagination.Forward` instead of `CursorBasedPagination.Forward` to the `extractPageInfo` closure.
+
+## Subscribing to results
+
+The `GraphQLQueryPager` class can fetch data, but the caller must subscribe to the results in order to receive the data. A `subscribe` method is provided which takes a closure that is called whenever the pager fetches a new page of data. The `subscribe` method is a convenience method that ensures that the closure is called on the main thread.
+
+```swift
+// Guaranteed to run on the main thread
+pager.subscribe { result in
+    switch result {
+    case .success(let data):
+        // Handle the data
+    case .failure(let error):
+        // Handle the error
+    }
+}
+```
+
+The `GraphQLQueryPager` is also a Combine `Publisher`. As such, all `Publisher` methods are available, such as `sink`, `assign`, `map`, `filter`, and so on.
+
+```swift
+// Can run on any thread
+pager.sink { result in
+    switch result {
+    case .success(let data):
+        // Handle the data
+    case .failure(let error):
+        // Handle the error
+    }
+}
+```
+
+## Fetching Data
+
+The `GraphQLQueryPager` class provides several methods to fetch paginated data: `fetch`, `refetch`, `loadNext`, `loadPrevious`, and `loadAll`.
+
+`fetch`: Fetches the first page of data. Must be called before `loadNext` or `loadPrevious` can be called.
+`refetch`: Cancels all ongoing fetch operations and resets the pager to its initial state. Fetches the first page of data.
+`loadNext`: Fetches the next page of data. Can only be called after `fetch` has been called.
+`loadPrevious`: Fetches the previous page of data. Can only be called after `fetch` has been called.
+`loadAll`: Fetches all pages of data. If no initial page is detected, it will first call `fetch` to fetch the first page of data. Will continue to fethc all pages until a `PageInfo` object indicates that there are no more pages to fetch. This function is compatible with forward, reverse, and bidirectional pagination.
```

Comment:
I think these need to be bulleted or else they will all appear as one paragraph. Also had a typo!

```suggestion
- `fetch`: Fetches the first page of data. Must be called before `loadNext` or `loadPrevious` can be called.
- `refetch`: Cancels all ongoing fetch operations and resets the pager to its initial state. Fetches the first page of data.
- `loadNext`: Fetches the next page of data. Can only be called after `fetch` has been called.
- `loadPrevious`: Fetches the previous page of data. Can only be called after `fetch` has been called.
- `loadAll`: Fetches all pages of data. If no initial page is detected, it will first call `fetch` to fetch the first page of data. Will continue to fetch all pages until a `PageInfo` object indicates that there are no more pages to fetch. This function is compatible with forward, reverse, and bidirectional pagination.
```